samples/web/content/apprtc/apprtc.py

```diff
@@ -366,12 +390,19 @@ def add_client_to_room(host, room_id, client_id, is_loopback):
       room.add_client(client_id, Client(is_initiator))
       if is_loopback:
         room.add_client(constants.LOOPBACK_CLIENT_ID, Client(False))
+      if allowed_clients is not None:
+        for allowed_client in allowed_clients:
+          room.add_allowed_client(allowed_client)
     else:
       is_initiator = False
       other_client = room.get_other_client(client_id)
       messages = other_client.messages
       room.add_client(client_id, Client(is_initiator))
       other_client.clear_messages()
+      # Check if the callee was the callee intended by the caller.
+      if not room.has_allowed_client(client_id):
+        error = constants.RESPONSE_INVALID_ROOM
```

Comment:
Maybe change allowed_clients to called_clients?

On Tue, Jan 27, 2015 at 4:11 PM, Chuck Hays <notifications@github.com>
wrote:

> In samples/web/content/apprtc/apprtc.py
> <https://github.com/GoogleChrome/webrtc/pull/436#discussion_r23655829>:
>
> >      else:
> >        is_initiator = False
> >        other_client = room.get_other_client(client_id)
> >        messages = other_client.messages
> >        room.add_client(client_id, Client(is_initiator))
> >        other_client.clear_messages()
> > +      # Check if the callee was the callee intended by the caller.
> > +      if not room.has_allowed_client(client_id):
> > +        error = constants.RESPONSE_INVALID_ROOM
>
> That is an interesting case. I was trying to avoid explicitly marking the
> rooms, but it looks like we will have to in some way. Instead of using
> allowed_clients = None, we will have to have a room type, or something
> similar.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/436/files#r23655829>.
>



-- 
Jiayang Liu